src/azul/service/manifest_service.py

```diff
@@ -256,8 +256,15 @@ class ManifestPartition:
 
     @classmethod
     def from_json(cls, partition: JSON) -> 'ManifestPartition':
-        # FIXME: Should convert search_after from list back to tuple
-        #        https://github.com/databiosphere/azul/issues/3291
+        part_etags = partition['part_etags']
+        search_after = partition['search_after']
+        part_etags = None if part_etags is None else tuple(part_etags)
+        search_after = None if search_after is None else tuple(search_after)
+        partition = {
+            **partition,
+            'part_etags': part_etags,
+            'search_after': search_after
+        }
```

Comment:
```suggestion
        for key in ('part_etags', 'search_after'):
            val = partition[key]
            if val is not None:
                partition[key] = tuple(val)
```
What do you think about this instead? It does have 2 dict lookups per key (one to get the value and one to set it), however it seems a bit more optimized than rebuilding the `partition` dictionary only to update those two values.

Reply:
Consider that the type of `partition` is `JSON`. When using your approach two things happen. One, using `partition[key] = …` raises a warning, something regarding `__setitem__` not being defined. So we can not use the mechanism you suggest unless we change the type of `partition`. Second, `JSON` doesn't define `tuple` as a type, only array (`list`), I used https://restfulapi.net/json-data-types/ as ref. So overriding the `partition: JSON` object with tuple values would be semantically incorrect. The approach taken considered the latter scenario, which is why the `partition` is regenerated from a `JSON` to a `dict`, it can then make use of `tuple` type. 
That said, I couldn't figure another mechanism for optimizing this generation, other than extracting the class attributes one by one.

Reply:
Interesting. In that case your approach is superior, however we might not want to overwrite the `partition` variable with a non `JSON` dictionary (containing tuples). Instead it could be directly unpacked into the `cls` constructor.

```
        return cls(**{
            **partition,
            'part_etags': part_etags,
            'search_after': search_after
        })
```